docs/book/how-to/build-pipelines/compose-pipelines.md

```diff
@@ -0,0 +1,41 @@
+---
+description: Compose your ZenML pipelines.
+---
+
+# Compose pipelines (reuse steps between pipelines)
+
+Sometimes it can be useful to extract some common functionality into separate functions
+in order to avoid code duplication. To facilitate this, ZenML allows you to compose your pipelines:
+
+```python
+from zenml import pipeline
+
+@pipeline
+def data_loading_pipeline(mode: str):
+    if mode == "train":
+        data = training_data_loader_step()
+    else:
+        data = test_data_loader_step()
+    
+    processed_data = preprocessing_step(data)
+    return processed_data
+
+
+@pipeline
+def training_pipeline():
+    training_data = data_loading_pipeline(mode="train")
+    model = training_step(data=training_data)
+    test_data = data_loading_pipeline(mode="test")
+    evaluation_step(model=model, data=test_data)
+```
+
+{% hint style="info" %}
```

Comment:
```suggestion
```

Here we are calling one pipeline from within another pipeline, so functionally the `data_loading_pipeline` is functioning as a step within the `training_pipeline` though when it runs it is executing as an entirely separate pipeline. These two pipelines will be visible as separate pipelines within the ZenML dashboard, but practically they are working together within the umbrella of the `training_pipeline`.

{% hint style="info" %}
```

Reply:
@strickvl not sure what you intended here, but I tried to fix it

Reply:
oh I think the extra backticks messed with the change. I just added some explanation. Let me take a look...